### DISCUSSION TITLE
How to auto register new EKS cluster into paralus dashboard automatically

### DISCUSSION THREAD

**Discussion by tppalani** [paralus/paralus#300] [Q&A] [answered]
### Describe the issue you're facing

-Hi Team, We are looking for some input from your team. Right now we are resgistering new clusters into the Paralus dashboard manually. Is there any way to perform the automation using the CI tool, which means that wherever a new cluster is getting onboarded into EKS, it should automatically discover the changes and register into the Paralus dashboard?

**Comment by niravparikh05** [comment, marked as answer]
Cluster auto discover feature is not yet present as part of paralus however there are couple of workarounds you can build e.g. using pctl cli or paralus tf to import cluster, download bootstrap yaml and apply that to your target cluster. These can be bundled as part of your cluster initialization or post hooks to your gitops. 

Secondly, use EKS launch templates to execute the initialization script using pctl. At a high level, the script needs to do the following

1) Create a system user with necessary roles, login using that system user and download the cli config ( manual one time step )
2) Download the latest pctl using tools like curl e.g. - curl -O 
https://github.com/paralus/cli/releases/download/v0.1.5/pctl_0.1.5_linux_amd64.tar.gz
3) Extract pctl tool
4) Invoke pctl create cluster with cli config. reference - https://www.paralus.io/docs/usage/cli
5) Invoke pctl download cluster bootstrap
6) apply bootstrap yaml as part of the cluster provisioning
